components/sensor/teleinfo.rst

```diff
@@ -97,6 +97,34 @@ Configuration variables:
 - **uart_id** (*Optional*, :ref:`config-id`): Manually specify the ID of the :ref:`UART Component <uart>` if you want
   to use multiple UART buses.
 
+
+.. warning::
+
+    UART hardware is recommanded for this sensor. The default Esphome's logger also use one hardware UART.
+    The ESP8266 only has two hardware UARTs, one of which is transmit-only (details in `Logger Component section <components/logger>`__).
```

Comment:
I would say something like:
```
On ESP8266, only 2 UARTs are available, one of it is TX only. This means only one UART is available for Teleinformation as RX is needed.
Since the only UART available for Teleinformation is already used by the logger, we need to either set the logger to use the second UART which is TX only or totally disable the logger.
A third solution would be to use another pin not dedicated to UART by the mean of Software UART, but unfortunately, this has been very unstable during development and it's not recommended.
To disable logger or make it use another UART, you can have a look to <path/to/uart/wiki>
```
We discussed it in #855